sorting/java/counting-sort.java

```diff
@@ -0,0 +1,62 @@
+public class CountingSort {
+
+    /**
+     * Public method to use the counting sort algortithm
+     * @param arr - the array in input
+     */
+    public void sort(int[] arr){
+
+        int[] sortedArray = new int[arr.length + 1];
+
+        // Search max element
+        int max = arr[0];
+        for(int i = 0; i < arr.length; i++){
+            if(arr[i] > max){
+                max = arr[i];
+            }
+        }
+
+        // Create counting array
+        int[] countArray = new int[max + 1];
+        for(int i = 0; i < countArray.length; i++){
+            countArray[i] = 0;
+        }
+
+        // Count elements in array
+        for(int i = 0; i < arr.length; i++){
+            countArray[arr[i]]++;
+        }
+
+
+        for(int i = 1; i < countArray.length; i++){
+            countArray[i] += countArray[i - 1];
+        }
+
+        for (int i = arr.length - 1; i >= 0; i--) {
+            sortedArray[countArray[arr[i]] - 1] = arr[i];
+            countArray[arr[i]]--;
+        }
+
+        System.arraycopy(sortedArray, 0, arr, 0, arr.length);
+    }
+
+    /**
+     * Main method for testing
+     * @param args
+     */
+    public void main(String[] args) throws Exception {
+        CountingSort count = new CountingSort();
```

Comment:
Hi @ValerioTrinca Welcome to DSA

- [ ] Please redefine the main function instead of: 
`public void main(String[] args)` to `public static void main(String[] args)`

- [ ] Please add a function or any way to print the sorted array after a successful sorting

Thanks

Reply:
Hi @TawfikYasser Thank you.
I corrected the code as suggested